### PR TITLE
Refactor Airtable calls + add login/lookup error handling

### PR DESCRIPTION
**Pull request by anniero98**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
- Updated everything to use the generator files - notably, this changes how we access objects returned from Airtable due to the record transformations being done by the helper functions in `request.js`
- Added error handling for both login and lookup (using enums)
- Fixed some conflicting rules between eslint and prettier
- Added a 'Testing Bypass' button to the login screen that takes you to checkout; defaults to "Jeffry Poa" and "Robin Hood" for Clerk/Customer respectively
- Added `prop-types` to do validation for props (specifically `props.navigation`) -- we should start doing this for all props (more important for components)

- Rewrote some things, some of which are:
## Refactor notes
- Refactored some state-modifying functions to use `prevState`: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-access-state-in-setstate.md
- Discovered this terrible thing with overriding format rules on save in VSCode is because Microsoft SUCKS https://github.com/prettier/prettier-vscode/issues/716
- Fix: set prettier config `printWidth` to 120 ):
- Decided to pass 'clerkName' through navigation instead of making an additional API call in 'CustomerLookupScreen'
- Removed the categories/filters for `CheckoutScreen`
- Updated positioning to be with `flex-direction` in the containers vs. `flex: #`

Next steps:
- We should add something to prompt the clerk to go back if the customer lookup was wrong (make sure they verify the names!)

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps

STYLE

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

Make sure everything still works (^:

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
image
image
image

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @wangannie @thumn 

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: lib/checkoutUtils.js [hunk 1]
```diff
@@ -1,156 +1,60 @@
...
+  updateCustomers,
+  updateTransactions
+} from './airtable/request';
+
+// Adds a field 'imageUrl' to products
 function createProductData(record) {
-  record = record.fields;
-  return {
-    name: record['Name'],
-    id: record.id,
```
Comment (thumn) [inline lib/checkoutUtils.js]: Oh is it that other fields like "name" and "points" are already accessibly by dot notation? :0

File: screens/ClerkLoginScreen.js [hunk 2]
```diff
@@ -67,33 +89,29 @@ export default class ClerkLoginScreen extends React.Component {
...
+            secureTextEntry
             onChangeText={text => this.setState({ password: text })}
             value={this.state.password}
           />
-          <SubmitButton
-            color="#008550"
-            title="Log In"
-            onPress={() => this.handleSubmit()}
-          />
+          <SubmitButton color="#008550" title="Log In" onPress={() => this.handleSubmit()} />
```
Comment (thumn) [inline screens/ClerkLoginScreen.js]: Should we use the `primaryGreen` color from Colors.js for this?

File: screens/CustomerLookupScreen.js [hunk 3]
```diff
@@ -73,12 +73,13 @@ export default class CustomerLookupScreen extends React.Component {
           onChangeText={number => this.setState({ phoneNumber: number })}
           value={this.state.phoneNumber}
         />
-        <SubmitButton
-          color="#008550"
-          title="Find Customer"
-          onPress={() => this.handleSubmit()}
-        />
+        <SubmitButton color="#008550" title="Find Customer" onPress={() => this.handleSubmit()} />
```
Comment (thumn) [inline screens/CustomerLookupScreen.js]: Same here for color!

File: lib/checkoutUtils.js (hunk 1 above)
Comment (anniero98) [reply lib/checkoutUtils.js]: Mhmm! One of the really nice things Aivant implemented with the generator functions is record transformations - ex. "Points Earned" --> `.pointsEarned`; and linked fields will automatically have a '.id' appended to them (so when updating lineItems, the attribute name is `productsPurchasedId`

you can see the object attributes in `schema.js` if you want to sanity-check 😂, but basically using `...record` says "keep everything already there" and then add on the new attributes

Can explain more in person!!

File: screens/ClerkLoginScreen.js (hunk 2 above)
Comment (wangannie) [reply screens/ClerkLoginScreen.js]: Eventually yeah but we can worry about this in the next styling PR.
